compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp

```diff
@@ -201,29 +373,58 @@ struct QuantizeDequantizeWeights final : public luci::CircleNodeMutableVisitor<b
       {
         auto circle_const = loco::must_cast<luci::CircleConst *>(circle_node);
 
+        // Find min/max per channel-wise
+        if (granularity)
+        {
+          std::vector<float> min;
+          std::vector<float> max;
+
+          cal_minmax_per_channel(circle_const, min, max);
+
+          std::vector<float> nudged_min(min.size());
+          std::vector<float> nudged_max(min.size());
+          std::vector<float> scaling_factor(min.size());
+          std::vector<int64_t> zp(min.size());
+
+          wquant_per_channel(circle_const, min, max, scaling_factor, zp, nudged_min, nudged_max);
+          wdequant_per_channel(circle_const, scaling_factor, nudged_min);
+
+          auto quantparam = std::make_unique<CircleQuantParam>();
+          for (std::vector<float>::size_type i = 0; i < min.size(); ++i)
+          {
+            quantparam->min.push_back(nudged_min[i]);
+            quantparam->max.push_back(nudged_max[i]);
+            quantparam->scale.push_back(scaling_factor[i]);
+            quantparam->zerop.push_back(zp[i]);
+          }
+          circle_node->quantparam(std::move(quantparam));
+        }
         // Find min/max per layer-wise
-        float min = std::numeric_limits<float>::max();
-        float max = std::numeric_limits<float>::min();
-        for (uint32_t i = 0; i < circle_const->size<loco::DataType::FLOAT32>(); i++)
+        else
         {
-          auto data = circle_const->at<loco::DataType::FLOAT32>(i);
-          min = data < min ? data : min;
-          max = data > max ? data : max;
+          float min = std::numeric_limits<float>::max();
+          float max = std::numeric_limits<float>::min();
```

Comment:
Please use numeric_limits<float>::lowest() instead of numeric_limits<float>::min()

Reply:
@jinevening , I wonder if there is a difference?

Reply:
> @jinevening , I wonder if there is a difference?

`numeric_limits<float>::min()` is the smallest **positive** value. 

`numeric_limits<float>::lowest()` is the smallest **negative** value.

I guessed the intention of the code is to use the smallest negative value.